### PR TITLE
add unit test

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?

*(For example: The Arctic Flink dataStream API has already supported this configuration: 'scan.startup.mode', but it is not available with Flink SQL. We should expose this configuration to users when they are using Flink SQL.)*

## Brief change log

*(For example:)*
  - *Add the 'scan.startup.mode' configuration to the ArcticScanContext in the flink modules*
  - *The flink 1.12 and 1.14 versions are affected.*
  - *Add the documentation in the flink chapter*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by shidayang** [comment]
Please add unit test for OptimizeCall and UseCall

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/ShowCall.java [hunk 1]
```diff
@@ -1,42 +1,45 @@
...
-  public String call(Context context) throws FullTableNameException {
-    if (this.namespaces == Namespaces.CATALOGS) {
-      return catalogManager.catalogs().stream().collect(Collectors.joining("\\n"));
-    }
-    if (context.getCatalog() == null) {
-      throw new FullTableNameException("Can not find catalog name, your can use 'USE ${catalog}' statement");
+  public String call(Context context) {
+    switch (this.namespaces) {
+      case CATALOGS:
+        return String.join("\\n", catalogManager.catalogs());
```
Comment (HuangFru) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/ShowCall.java]: use "\n" instead of "\\\n" to represent line break?

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/handler/impl/OptimizeManagerHandler.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ public OptimizerDescriptor registerOptimizer(OptimizerRegisterInfo registerInfo)
   @Override
   public void stopOptimize(TableIdentifier tableIdentifier) throws OperationErrorException, TException {
     try {
-      ServiceContainer.getOptimizerService().stopOptimize(tableIdentifier);
+      ServiceContainer.getOptimizeService().stopOptimize(com.netease.arctic.table.TableIdentifier.of(tableIdentifier));
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/handler/impl/OptimizeManagerHandler.java]: Add a method to TableIdentifier to generator com.netease.arctic.table.TableIdentifier

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/handler/impl/OptimizeManagerHandler.java [hunk 3]
```diff
@@ -71,7 +71,7 @@ public void stopOptimize(TableIdentifier tableIdentifier) throws OperationErrorE
   @Override
   public void startOptimize(TableIdentifier tableIdentifier) throws OperationErrorException, TException {
     try {
-      ServiceContainer.getOptimizerService().startOptimize(tableIdentifier);
+      ServiceContainer.getOptimizeService().startOptimize(com.netease.arctic.table.TableIdentifier.of(tableIdentifier));
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/handler/impl/OptimizeManagerHandler.java]: Also

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/RefreshCall.java [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
 
 public class RefreshCall implements CallCommand {
 
-  private AmsClient client;
+  private ArcticTableMetastore.Iface client;
   private String tablePath;
+  private static final String OK_CODE = "OK";
 
-  public RefreshCall(AmsClient client, String tablePath) {
+  public RefreshCall(ArcticTableMetastore.Iface client, String tablePath) {
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/RefreshCall.java:14]: Make ArcticTableMetastore.Iface to a Interface

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/RefreshCallGenerator.java [hunk 5]
```diff
@@ -1,13 +1,11 @@
...
-import com.netease.arctic.AmsClient;
+import com.netease.arctic.ams.api.ArcticTableMetastore;
 import com.netease.arctic.ams.api.client.AmsClientPools;
-import com.netease.arctic.catalog.ArcticCatalog;
-import com.netease.arctic.catalog.CatalogLoader;
 
 public class RefreshCallGenerator {
 
-  private AmsClient client;
+  private ArcticTableMetastore.Iface client;
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/RefreshCallGenerator.java]: Also

File: ams/ams-server/src/test/java/com/netease/arctic/ams/server/repair/TestRepairEnv.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+import org.apache.iceberg.types.Types;
+import org.assertj.core.util.Lists;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+
+import static com.netease.arctic.ams.api.properties.CatalogMetaProperties.CATALOG_TYPE_HADOOP;
+
+public class TestRepairEnv {
```
Comment (shidayang) [inline ams/ams-server/src/test/java/com/netease/arctic/ams/server/repair/TestRepairEnv.java]: This calss is useless

**Review by shidayang — CHANGES_REQUESTED** [review]

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/ShowCall.java [hunk 7]
```diff
@@ -1,42 +1,45 @@
...
 
-  private CatalogManager catalogManager;
   private Namespaces namespaces;
+  private CatalogManager catalogManager;
 
-  public ShowCall(CatalogManager catalogManager, Namespaces namespaces) {
-    this.catalogManager = catalogManager;
+  public ShowCall(String amsAddress, Namespaces namespaces) {
     this.namespaces = namespaces;
+    this.catalogManager = new CatalogManager(amsAddress);
```
Comment (HuangFru) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/ShowCall.java]: Maybe creating CatalogManager in Generator is better.

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/handler/impl/OptimizeManagerHandler.java (hunk 2 above)
Comment (hameizi) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/handler/impl/OptimizeManagerHandler.java]: The TableIdentifier is a class generated by thrift, and you can't customize methods

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/ShowCall.java (hunk 7 above)
Comment (shidayang) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/ShowCall.java]: @HuangFru I agree with you

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/handler/impl/OptimizeManagerHandler.java (hunk 2 above)
Comment (shidayang) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/handler/impl/OptimizeManagerHandler.java:65]: Makr a util?

**Review by shidayang — CHANGES_REQUESTED** [review]

**Review by shidayang — APPROVED** [review]
